### PR TITLE
revset: add literal:"string" pattern syntax

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/revset.rs [hunk 1]
```diff
@@ -1269,8 +1272,51 @@ fn parse_function_argument_to_string_pattern(
...
+            let RevsetExpression::CommitRef(RevsetCommitRef::Symbol(kind)) = roots.as_ref() else {
+                return Err(make_type_error());
+            };
+            let RevsetExpression::CommitRef(RevsetCommitRef::Symbol(needle)) = heads.as_ref()
+            else {
+                return Err(make_type_error());
+            };
+            match kind.as_ref() {
+                "literal" => StringPattern::Literal(needle.clone()),
+                // TODO: maybe add explicit kind for substring match?
```
Comment (martinvonz) [inline lib/src/revset.rs:1319]: Yes, that would seem consistent. We can also reevaluate if we want the default to be substring matching.

**Review by martinvonz — APPROVED** [review]
Thank you!

File: docs/revsets.md [hunk 2]
```diff
@@ -135,6 +135,13 @@ revsets (expressions) as arguments.
 * `present(x)`: Same as `x`, but evaluated to `none()` if any of the commits
   in `x` doesn't exist (e.g. is an unknown branch name.)
 
+## String patterns
+
+Functions that perform string matching support the following pattern syntax.
+
+* `"needle"`: The haystack contains `needle`.
```
Comment (necauqua) [inline docs/revsets.md:142]: Ok I know the saying, and it even directly translates to the same saying in my native language, but still I feel like technical "The string contains a substring `not needle`" would be sliiightly less confusing to the reader, especially as there is zero explicit association between "haystack" (suppose I don't know what that word means) and the string we're trying to match.

Comment (joyously) [reply docs/revsets.md:142]: That doesn't make sense to me...
How about "container to search" and "what to search for"?

Comment (emilykfox) [reply docs/revsets.md:142]: Maybe:
* `"substring"`: Matches strings that contain `substring`.
* `literal:"string"`:  Matches strings exactly equal to `string`.

There's only one other public-facing use of `needle` in the docs as far as I can tell (in [docs/templates.md](https://github.com/martinvonz/jj/blob/313670d3c22ee4ddb715cb4e2d736bedcaa4368f/docs/templates.md?plain=1#L140)), so I don't see a strong motivation to keep using the word.

Also, many of the uses of `(something_)pattern` above still use the verb "contains" or "is" when "matches" and "matching" would be more appropriate with the new pattern syntax.

Comment (yuja) [reply docs/revsets.md]: > * `"substring"`: Matches strings that contain `substring`.
> * `literal:"string"`:  Matches strings exactly equal to `string`.

Much better! Thanks for the suggestion.

**Pull request merged**